docs/source/workflow.rst

```diff
@@ -231,6 +232,37 @@ Valid options are:
 * ``check_haproxy``: Whether to check the local haproxy to make sure this task
   has been registered and discovered.
 
+Draining
+--------
+Draining is the process to stop instances of an old service from receiving
+traffic. PaaSTA supports pluggable drain methods for service authors to mark
+services up and down in their environments.
+
+Current master has three draining methods:
+
+* `noop <generated/paasta_tools.drain_lib.html#drain_lib.NoopDrainMethod>`_ - This draining method skips
+  draining completely. Service instances are killed as needed.
+
+* `test <generated/paasta_tools.drain_lib.html#drain_lib.TestDrainMethod>`_ - This draining method uses
+  class variables to keep track of instances that are marked down to drain and
+  instances that have stopped receiving traffic.
+
+* `hacheck <generated/paasta_tools.drain_lib.html#drain_lib.HacheckDrainMethod>`_ - `hacheck <https://github.com/Yelp/hacheck>`_ is
+  used at Yelp to provide APIs to query and change state of a service instance.
+  The hacheck draining method requests hacheck to mark down an instance. HAProxy
+  checks with hacheck periodically to keep its view of instance state up-to-date.
+  The hacheck draining method will wait for a configurable ``delay`` to make sure
+  HAProxy has the update before considering safe to kill an instance. Note that
+  the hacheck draining method sets an expiration when marking an instance down on
+  hacheck. hacheck will drop the down state if it receives a status query after
+  expiration.
+
+**What is the rationale behind the expiration logic? We have seen boucing being
```

Comment:
The bold part is actually a question for reviewers regarding the draining expiration logic.